src/main/java/de/tum/in/www1/artemis/service/plagiarism/PlagiarismService.java

```diff
@@ -163,4 +171,53 @@ private boolean userForSubmissionDeleted(Submission submission) {
         }
         return true; // if the user is not found, we assume that the user has been deleted
     }
+
+    /**
+     * filter a participation based on the last submission and make sure the last result - if available - has a minimum score - in case this is larger than 0
+     *
+     * @param minimumScore the minimum score
+     * @return a predicate that can be used in streams for filtering
+     */
+    @NotNull
+    public static Predicate<ProgrammingExerciseParticipation> filterParticipationMinimumScore(int minimumScore) {
+        return participation -> {
+            Submission submission = participation.findLatestSubmission().orElse(null);
+            // filter empty submissions
+            if (submission == null) {
+                return false;
+            }
+            return hasMinimumScore(submission, minimumScore);
+        };
+    }
+
+    /**
+     * filter a submission so that the last result - if available - has a minimum score - in case this is larger than 0
+     *
+     * @param submission   the submission which latest result should be tested for filtering
+     * @param minimumScore the minimum score
+     * @return a predicate that can be used in streams for filtering
+     */
+    public static boolean hasMinimumScore(Submission submission, int minimumScore) {
+        return minimumScore == 0
+                || submission.getLatestResult() != null && submission.getLatestResult().getScore() != null && submission.getLatestResult().getScore() >= minimumScore;
+    }
+
+    /**
+     * filter a student participation so that it includes a student that is not at least a tutor, in case of team participations no filter is applied
+     *
+     * @return a predicate that can be used in streams for filtering
+     */
+    @NotNull
+    public Predicate<StudentParticipation> filterForStudents() {
+        return participation -> {
+            // this filter excludes individual participations that are not from students
+            if (participation.getStudent().isPresent()) {
+                return !authCheckService.isAtLeastTeachingAssistantForExercise(participation.getExercise(), participation.getStudent().get());
+            }
+            // however, we make sure that team participations are included (in the unlikely case, neither student and team is defined, we filter)
+            else {
+                return participation.getTeam().isPresent();
```

Comment:
Is `team` fetched from db?

Reply:
yes. the problem is usually the connection between team and students, not between participation and team.